Update MCU LTO recommendation language (#498)
memfault/memfault-docs: docs/mcu/faq.mdx

### Link Time Optimization (LTO) `-flto`

We strongly recommend _against_ using Link Time Optimization (LTO) in the

> programs you want to be able to debug. See more explanation now:
> 
> <https://interrupt.memfault.com/blog/best-and-worst-gcc-clang-compiler-flags#-flto>

Reviewer: ```suggestion
programs you want to be able to debug. See this [Interrupt post](https://interrupt.memfault.com/blog/best-and-worst-gcc-clang-compiler-flags#-flto) for more info.

<https://interrupt.memfault.com/blog/best-and-worst-gcc-clang-compiler-flags#-flto>
```
